### ISSUE TITLE
nexus_repository resource - http_client block is optional in the doc, but not in reality

### ISSUE THREAD

**Issue by mpailloncy** [datadrivers/terraform-provider-nexus#141]
**Describe the bug**

The [documentation for nexus_repository](https://registry.terraform.io/providers/datadrivers/nexus/latest/docs/resources/repository#http_client) states that _http_client_ block is optional, while it's actually required by the provider.
 

**To Reproduce**
1. Given this _nexus_repository_ definition:

````
resource "nexus_repository" "docker_quay_proxy" {
  name   = "docker-quay"
  format = "docker"
  type   = "proxy"

  proxy {
    remote_url       = "https://quay.io"
    content_max_age  = "1440"
    metadata_max_age = "1440"
  }

  docker {
    force_basic_auth = false
    http_port        = 0
    https_port       = 0
    v1enabled        = false
  }

  docker_proxy {
    index_type = "REGISTRY"
  }

  negative_cache {
    enabled = false
  }

  storage {
    blob_store_name                = "default"
    strict_content_type_validation = true
  }

}

````

2. Try to plan/apply it
3. An error like this one is displayed on "apply":

````
╷
│ Error: could not update repository 'docker-quay': HTTP: 400, [ {
│   "id" : "PARAMETER httpClient",
│   "message" : "must not be null"
│ } ]
│ 
│   with nexus_repository.docker_quay_proxy,
│   on repositories-docker.tf line 42, in resource "nexus_repository" "docker_quay_proxy":
│   42: resource "nexus_repository" "docker_quay_proxy" {
│ 
╵
````

Adding an empty block `http_client {}` in the _nexus_repository_ def resolves the problem.

**Expected behavior**

_http_block_ block should be optional

**Comment by anmoel** [comment]
Hi @mpailloncy ,

thanks for your report.
The nexus api required the http_client block. 
We will fix the documentation and in the near future we will split all repository types into their own resources.
This way it is better documented and we can deal with it better.

regards
André

**Comment by anmoel** [comment]
this feature is implemented in version v1.15.0
